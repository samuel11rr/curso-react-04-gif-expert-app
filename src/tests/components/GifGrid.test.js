import React from 'react'
import '@testing-library/jest-dom'
import { useFetchGifs } from '../../hooks/useFetchGifs';
const { shallow } = require("enzyme");
const { GifGrid } = require("../../components/GifGrid");
jest.mock('../../hooks/useFetchGifs');


describe('Pruebas en el <GifGrid />', () => {
  const category = 'One punch';

  test('Debe mostrarse correctamente', () => {
    useFetchGifs.mockReturnValue({
      data: [],
      loading: true
    });

    const wrapper = shallow( <GifGrid category={ category } /> );
    expect( wrapper ).toMatchSnapshot();
  });
  

  test('Debe mostrar items cuando se cargan imagenes useFetchGifs', () => {
    const gifs = [{
      id: 'ABC',
      url: 'https://localhost/prueba.png',
      title: 'Pruebas'
    }]

    useFetchGifs.mockReturnValue({
      data: gifs,
      loading: false
    });

    const wrapper = shallow( <GifGrid category={ category } /> );
    
    expect( wrapper ).toMatchSnapshot();
    expect( wrapper.find('p').exists() ).toBe(false);
    expect( wrapper.find('GifGridItem').length ).toBe( gifs.length );
  });
  
});
