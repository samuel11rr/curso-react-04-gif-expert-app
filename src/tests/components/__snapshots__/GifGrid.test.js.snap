// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el <GifGrid /> Debe mostrar items cuando se cargan imagenes useFetchGifs 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    One punch
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="Pruebas"
      url="https://localhost/prueba.png"
    />
  </div>
</Fragment>
`;

exports[`Pruebas en el <GifGrid /> Debe mostrarse correctamente 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    One punch
  </h3>
  <p
    className="card animate__animated animate__flash"
  >
    Loading
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
